File: 820-short-encoding-of-words/820-short-encoding-of-words.py
class Solution:
    def minimumLengthEncoding(self, words: List[str]) -> int:
        mySet = set(words)
        
        for word in words:
            for i in range(1, len(word)):
                if word[i: ] in mySet:
                    mySet.remove(word[i: ])
                
        ans = 0
        
        for word in mySet:
            ans += len(word) + 1
            
        return ans
        

File: 820-short-encoding-of-words/820-short-encoding-of-words.cpp
class Solution {
public:
    int minimumLengthEncoding(vector<string>& words) {
        return daily(words);
    }
private:
    int daily(vector<string>& words) {
        unordered_set<string> mySet(words.begin(), words.end());
        
        for(auto word: words) {
            for(int i = 1; i < word.length(); ++i) 
                mySet.erase(word.substr(i));
        }
        int ans = 0;
        
        for(string word: mySet)
            ans += word.length() + 1;
        return ans;
    }
};